src/Workspaces/CSharp/Portable/LanguageServices/CSharpSyntaxFactsService.cs

```diff
@@ -1343,6 +1343,25 @@ public SeparatedSyntaxList<SyntaxNode> GetArgumentsOfObjectCreationExpression(Sy
         public SeparatedSyntaxList<SyntaxNode> GetArgumentsOfArgumentList(SyntaxNode argumentList)
             => (argumentList as ArgumentListSyntax)?.Arguments ?? default(SeparatedSyntaxList<SyntaxNode>);
 
+        public SeparatedSyntaxList<SyntaxNode> GetTypeArgumentsOfInvocationExpression(SyntaxNode invocationExpression)
+        {
+            if (invocationExpression is InvocationExpressionSyntax invocationExpressionSyntax)
+            {
+                switch (invocationExpressionSyntax.Expression)
+                {
+                    case GenericNameSyntax genericName:
+                        return genericName.TypeArgumentList.Arguments;
+                    case MemberAccessExpressionSyntax memberAccessExpression when memberAccessExpression.Name is GenericNameSyntax genericName:
+                        return genericName.TypeArgumentList.Arguments;
```

Comment:
1. what happens if you have a ```foo?.Bar<X>``` ```foo.bar?.Baz<X>``` or ```foo?.bar.baz<X>``` ?
2. Syntax facts are for unifying concepts over VB and C#, not for generalized deconstruction APIs.  So, in this case, you should remove this helper and move it to the AbstractAddParameter code.  It would grab the ExpressionOfInvocationExpression then do the tests about GenericName/MemberAccess, and decompose things there.